src/data/scripts/generate_historic_stats.py

```diff
@@ -0,0 +1,144 @@
+"""
+to run this script
+python3 -m src.data.scripts.generate_historic_stats
```

Comment:
Also all other steps log a standard completion notice, see `log_step_completion` in any of the other files. That keeps things consistent!

<img width="719" height="477" alt="image" src="https://github.com/user-attachments/assets/b388cdd9-517e-44f4-8155-f061d9effb23" />